pkg/storage/store.go

```diff
@@ -56,6 +62,16 @@ func NewStore(cfg Config, storeCfg chunk.StoreConfig, schemaCfg chunk.SchemaConf
 	}, nil
 }
 
+// NewTableClient creates a TableClient for managing tables for index/chunk store.
+// ToDo: Add support in Cortex for registering custom table client like index client.
+func NewTableClient(name string, cfg Config) (chunk.TableClient, error) {
```

Comment:
This is supported now I think.

Reply:
It is not supported yet. We have updated cortex version in vendor to `v1.0.0` while my changes are added after that. I will have to update it again to latest master. I think we should do a follow-up commit for that.